presto-product-tests/src/main/java/io/prestosql/tests/hive/TestAllDatatypesFromHiveConnector.java

```diff
@@ -398,17 +398,26 @@ public void testSelectAllDatatypesParquetFile()
                         234.567,
                         new BigDecimal("346"),
                         new BigDecimal("345.67800"),
-                        getHiveVersionMajor() < 3
-                                ? Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 12, 15, 35, 123_000_000))
+                        isHiveWithBrokenTimestampSemantics()
                                 // TODO (https://github.com/prestosql/presto/issues/1218) requires https://issues.apache.org/jira/browse/HIVE-21002
-                                : Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 18, 0, 35, 123_000_000)),
+                                ? Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 18, 0, 35, 123_000_000))
+                                : Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 12, 15, 35, 123_000_000)),
                         "ala ma kota",
                         "ala ma kot",
                         "ala ma    ",
                         true,
                         "kot binarny".getBytes(UTF_8)));
     }
 
+    private boolean isHiveWithBrokenTimestampSemantics()
+    {
+        // In 3.1.0 timestamp semantics in hive changed in backward incompatible way, which was fixed in 3.1.2
+        // https://issues.apache.org/jira/browse/HIVE-21002
```

Comment:
That's not exact. 

> In 3.1.0 timestamp semantics in hive changed in backward incompatible way

True, https://issues.apache.org/jira/browse/HIVE-12192

> which was fixed in 3.1.2

portion thereof, not all of it